src/utils/gitHubUtils.ts

```diff
@@ -167,3 +169,38 @@ export function isUser(orgData: UsersGetAuthenticatedResponseData | OrgForAuthen
     // if there's no orgData, just assume that it's a user (but this shouldn't happen)
     return !!orgData && 'type' in orgData && orgData.type === 'User';
 }
+
+export async function getGitHubTree(repositoryUrl: string, branch: string): Promise<GitTreeData[]> {
+    const octokitClient: Octokit = await createOctokitClient();
+    const { owner, name } = getRepoFullname(repositoryUrl);
+    const branchRes: OctokitResponse<ReposGetBranchResponseData> = await octokitClient.repos.getBranch({ owner, repo: name, branch });
+    const getTreeRes: OctokitResponse<GitGetTreeResponseData> = await octokitClient.git.getTree({ owner, repo: name, tree_sha: branchRes.data.commit.sha, recursive: 'true' });
+
+    return getTreeRes.data.tree.filter(file => file.type === 'tree').sort((f1, f2) => {
```

Comment:
Two things - 
1. I'd prefer you create a function `getFolderDepth` to avoid copied code
1. When using `sort` it doesn't have to be exactly 1 or -1. It just has to be a positive or negative number. Thus it's more traditional to just subtract the two values. This is shorter and more accurately handles the "0" case.

```
// sort descending by the depth of subfolder
return getTreeRes.data.tree.filter(file => file.type === 'tree').sort((f1, f2) => {
    // tslint:disable-next-line: strict-boolean-expressions
    function getFolderDepth(path: string): number { return (path.match(/\//g) || []).length; }
    return getFolderDepth(f1.path) - getFolderDepth(f2.path);
});
```